scene/gui/text_edit.h

```diff
@@ -41,6 +41,8 @@
 class TextEdit : public Control {
 	GDCLASS(TextEdit, Control);
 
+	friend class CodeHighlighter;
```

Comment:
Not a fan of this, as it seems a bit too permissive for accessing a single colour.

Could we have  `TextEdit` pass it through a setter instead?

Reply:
I'm not familiar enough with how `CodeHighlighter` works to know how to do that, but we can create a getter on `TextEdit` instead of providing private access. Though classes seemed to be closely related here, so I went with the friendship.

Reply:
A getter would work great to!